File: app/discover/fetch_region_object_types.rb
require_relative 'db_access'

class FetchRegionObjectTypes < DbAccess
  
  def get(parent)
    ret = {
      "type" => "region object type",
      "id" => "",
      "parent" => parent, 
      "rows" => [
        {"id" => "projects root", "text" => "projects", "descendants" => 1},
        {"id" => "aggregates root", "text" => "aggregates", "descendants" => 1},
        {"id" => "availability zones root", "text" => "availability zones", "descendants" => 1}
      ]
    }
    return ret
  end
  
end

File: app/discover/fetch_region_object_types.py
from fetcher import Fetcher


class FetchRegionObjectTypes(Fetcher):
    def __init(self):
        pass

    def get(self, parent):
        ret = {
            "id": "",
            "parent": parent,
            "rows": [
                {
                    "id": "aggregates_root",
                    "type": "aggregates_folder",
                    "text": "Aggregates"
                },
                {
                    "id": "availability_zones_root",
                    "type": "availability_zones_folder",
                    "text": "Availability Zones"
                },
                {
                    "id": "network_agents_root",
                    "type": "network_agents_folder",
                    "text": "network Agents"
                }
            ]
        }
        return ret
